lexos/static/js/scripts_scrub.js

```diff
@@ -1,267 +1,309 @@
+    $('#error-modal').modal()
+    return
+  }
+
+  $('#status-prepare').css({'visibility': 'visible'})
 
-  $(document).on('click', '#set-tags-button', function (event) {
-    if ($('#allTags')) {
-      var allTags = ($('#allTags')[0].value)
-      allTags = JSON.stringify(allTags)
-      $.ajax({
-        type: 'POST',
-        url: '/setAllTagsTable',
-        data: allTags,
-        'contentType': 'application/json; charset=utf-8',
-        'dataType': 'json',
-        beforeSend: function () {
-          $('<p/>', {
-            id: 'xmlModalStatus',
-            style: 'width:100px;margin:50px auto;z-index:1000;'
-          }).appendTo('#xmlModalBody')
-          $('#xmlModalStatus').append('<img src="/static/images/loading_icon.svg?ver=2.5" alt="Loading..."/>')
-        },
-        success: function (response) {
-          var selection = $('#allTags option:selected').val()
-          $('#tagTable').empty().remove()
-          var t = '<table id="tagTable" class="table table-condensed table-striped table-bordered"></table>'
-          $('#xmlModalBody').append(t)
-          var select = '<select id="allTags" style="margin-top:3px;margin-right:5px;">'
-          select += '<option value="remove-tag,allTags">Remove Tag Only</option>'
-          select += '<option value="remove-element,allTags">Remove Element and All Its Contents</option>'
-          select += '<option value="replace-element,allTags">Replace Element and Its Contents with Attribute Value</option>'
-          select += '<option value="leave-alone,allTags">Leave Tag Alone</option>'
-          select += '</select>'
-          select += '<button id="set-tags-button" type="button" class="btn btn-primary"">Set All</button>'
-          $('#tagTable').append('<thead><tr><th>Element</th><th>Action</th><th>' + select + '</th></tr></thead>')
-          $('#tagTable').append('<tbody></tbody>')
-		      $('#tagTable tbody').append(response)
-          $('#xmlModalStatus').remove()
-          $("#allTags option[value='"+selection+"']").prop("selected", true)
-	    		var value = $('#myselect option:selected').val()
-	    		var text = $('#myselect option:selected').text()
-        },
-        error: function (jqXHR, textStatus, errorThrown) {
-          console.log('Error: ' + errorThrown)
-        }
-      })
+  $('#formAction').val(action)
+  const formData = new FormData($('form')[0])
+
+  $.ajax({
+    url: '/doScrubbing',
+    type: 'POST',
+    processData: false, // important
+    contentType: false, // important
+    data: formData,
+    error: function (jqXHR, textStatus, errorThrown) {
+      $('#error-modal-message').html('Lexos could not apply the scrubbing actions.')
+      $('#error-modal').modal()
+      console.log('bad: ' + textStatus + ': ' + errorThrown)
     }
+  }).done(function (response) {
+    response = JSON.parse(response)
+    $('#preview-body').empty()
+    $.each(response['data'], function () {
+      const fileName = $(this)[1]
```

Comment:
Avoid using `this`, and index

Reply:
I'm not sure if there is a good way to rewrite this section without using `this` and index. This section gets the file names and previews for each active file and adds them to the DOM. This information is stored in an array once the JSON is parsed so I am not sure how else we can do this without using either `this` or index.

Reply:
In this case if `function (i, val)` is used, `i` is of course the index of each value but `val` is another array. How can I access specific indices of `val` without using `[1]` or `[3]`? I'm assuming that we do not want to use another `$.each()` within the one that is already there.

Reply:
That seems like a legacy problem of the back end. Typically in this case, we don't use number index, but use name index. Instead of `[1]`, we usually use stuff like `['filename']` for better readability.

In this case you can keep the number indexing.